src/coreclr/jit/assertionprop.cpp

```diff
@@ -4648,14 +4648,28 @@ class AssertionPropFlowCallback
     }
 
     // During merge, perform the actual merging of the predecessor's (since this is a forward analysis) dataflow flags.
-    void Merge(BasicBlock* block, BasicBlock* predBlock, flowList* preds)
+    void Merge(BasicBlock* block, BasicBlock* predBlock, unsigned dupCount)
     {
-        ASSERT_TP pAssertionOut = ((predBlock->bbJumpKind == BBJ_COND) && (predBlock->bbJumpDest == block))
-                                      ? mJumpDestOut[predBlock->bbNum]
-                                      : predBlock->bbAssertionOut;
+        ASSERT_TP pAssertionOut;
+
+        if (predBlock->bbJumpKind == BBJ_COND && (predBlock->bbJumpDest == block))
+        {
+            pAssertionOut = mJumpDestOut[predBlock->bbNum];
+
+            if (dupCount > 1)
```

Comment:
This deserves a comment and some extra jit dump output, since it is an unusual case.

Reply:
Agree. Added.